api_client/src/lib.rs

```diff
@@ -3,41 +3,27 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-use std::fmt;
 use std::io::{Read, Write};
 use std::os::unix::io::RawFd;
+use thiserror::Error;
 use vmm_sys_util::sock_ctrl_msg::ScmSocket;
 
-#[derive(Debug)]
+#[derive(Debug, Error)]
 pub enum Error {
+    #[error("Error writing to or reading from HTTP socket: {0}")]
     Socket(std::io::Error),
+    #[error("Error writing to or reading from HTTP socket: {0}")]
```

Comment:
I know you copied from the `Display` implementation but this error is about sending file descriptors. Please fix as a separate commit.